### PR TITLE
First integration of new modals library

### PR DESCRIPTION
**Pull request by dmohs**
https://broadinstitute.atlassian.net/browse/GAWB-2157

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/create.cljs [hunk 1]
```diff
@@ -93,8 +95,11 @@
 
 (react/defc Button
   {:render
-   (fn [{:keys [props]}]
+   (fn [{:keys [props state]}]
      [:div {:style {:display "inline"}}
+      (when (:modal? @state)
+        [CreateDialog (merge (select-keys props [:billing-projects])
+                             {:dismiss #(swap! state dissoc :modal?)})])
```
Comment (dmohs) [inline src/cljs/main/broadfcui/page/workspace/create.cljs:102]: Besides slightly improved functionality, this is the primary difference: We can now use modals similarly to normal React elements.

File: project.clj [hunk 2]
```diff
@@ -4,6 +4,7 @@
    [inflections "0.9.14"]
    [dmohs/react "1.1.2+15.5.4-0"]
    [frankiesardo/linked "1.2.9"]
+   [org.broadinstitute/react-cljs-modal "20170609"]
```
Comment (zarsky-broad) [inline project.clj]: Have you considered versioning with dashes, i.e. `2017-06-09`? Easier to parse.

File: src/cljs/main/broadfcui/components/modals.cljs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+(ns broadfcui.components.modals
+  (:require
+   [broadfcui.common :as common]
+   [broadfcui.common.components :as comps]
+   [broadfcui.common.style :as style]
+   [dmohs.react :as r]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/components/modals.cljs]: Hoist this to match other `require`s

File: src/cljs/main/broadfcui/components/modals.cljs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+(r/defc OKCancelForm
+  {:get-default-props
+   (fn []
+     {:show-cancel? true
+      :show-close? true})
+   :render
+   (fn [{:keys [this props]}]
+     (let [{:keys [header content dismiss ok-button show-cancel? cancel-text show-close?]} props
+           cancel-text (or cancel-text "Cancel")]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/components/modals.cljs]: Why not just have a default prop for `cancel-text`?

File: src/cljs/main/broadfcui/components/modals.cljs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+       (modal/render
+        {:content
+         [:div {}
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/components/modals.cljs]: Can we deal with all of the weird percentages and arbitrary non-rem pixel values while we're in here?

File: src/cljs/main/broadfcui/components/modals.cljs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                     :style {:marginRight (when ok-button 27) :marginTop 2
+                             :display "inline-block"
+                             :fontSize "106%" :fontWeight 500 :textDecoration "none"
+                             :color (:button-primary style/colors)}
+                     :href "javascript:;"
+                     :onClick dismiss
+                     :onKeyDown (common/create-key-handler [:space :enter] dismiss)}
+                 cancel-text])
+              (when ok-button
+                (cond (string? ok-button) [comps/Button {:text ok-button :ref "ok-button" :class-name "ok-button" :onClick dismiss}]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/components/modals.cljs]: What does this class name do? It appears to be totally useless, presumably left over. Let's kill it while we're in here.

**Review by zarsky-broad — APPROVED** [review]

File: project.clj (hunk 2 above)
Comment (dmohs) [reply project.clj]: I see the benefit there, but this format is compatible with SemVer. Maybe worth the loss of readability?

Comment (zarsky-broad) [reply project.clj]: `2016.06.09`, then?

**Comment by dmohs** [comment]
@MattPutnam I'm going to hold-off on merging this until you've had a chance to take a look.

**Review by MattPutnam — APPROVED** [review]
👍

**Pull request merged**